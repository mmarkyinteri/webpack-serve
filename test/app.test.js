const http = require('http');
const net = require('net');

const weblog = require('webpack-log');

const { getApp } = require('../lib/app');
const eventbus = require('../lib/bus');
const { getCompiler } = require('../lib/compiler');
const { getOptions } = require('../lib/options');

/* eslint-disable no-param-reassign */

describe('app', () => {
  weblog({
    id: 'webpack-serve',
    level: 'silent',
    name: 'serve',
  });

  test('getApp', () => {
    const config = require('./fixtures/basic/webpack.config');
    const content = ['./batman', './superman'];
    const logLevel = 'silent';
    const options = {
      bus: eventbus({}),
      content,
      devMiddleware: { publicPath: '/', logLevel },
      hotClient: { logLevel },
      host: 'localhost',
      port: 0,
      protocol: 'http',
    };
    options.compiler = getCompiler([config], options);
    const app = getApp(options);

    expect(app).toMatchSnapshot();

    return app.start().then(([, , server]) => {
      expect(server).toBeInstanceOf(net.Server);
      expect(server).toBeInstanceOf(http.Server);
      expect(server.constructor.name).toMatchSnapshot();
      return new Promise((resolve) => {
        setTimeout(() => app.stop(resolve), 500);
      });
    });
  });

  test('getApp + getOptions', () => {
    const argv = {
      config: './test/fixtures/basic/webpack.config.js',
      logLevel: 'silent',
    };
    const opts = {};

    return getOptions(argv, opts).then((result) => {
      const { configs, options } = result;

      expect(configs).toMatchSnapshot();
      expect(options).toMatchSnapshot();

      options.bus = eventbus({});
      options.compiler = getCompiler(configs, options);
      const app = getApp(options);

      return app.start().then(([, , server]) => {
        expect(server).toBeInstanceOf(net.Server);
        expect(server).toBeInstanceOf(http.Server);
        expect(server.constructor.name).toMatchSnapshot();
        return new Promise((resolve) => {
          setTimeout(() => app.stop(resolve), 500);
        });
      });
    });
  });
});
