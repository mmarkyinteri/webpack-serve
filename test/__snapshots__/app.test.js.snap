// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app getApp + getOptions 1`] = `
Array [
  Object {
    "context": "<PROJECT_ROOT>/test/fixtures/basic",
    "devtool": "source-map",
    "entry": Array [
      "./app.js",
    ],
    "output": Object {
      "filename": "./output.js",
      "path": "<PROJECT_ROOT>/test/fixtures/basic",
    },
    "plugins": Array [
      TimeFixPlugin {
        "watchOffset": 11000,
      },
      NamedModulesPlugin {
        "options": Object {},
      },
    ],
    "serve": Object {},
  },
]
`;

exports[`app getApp + getOptions 2`] = `
Object {
  "clipboard": true,
  "compiler": null,
  "content": Array [
    "<PROJECT_ROOT>/test/fixtures/basic",
  ],
  "devMiddleware": Object {
    "logLevel": "silent",
    "publicPath": "/",
  },
  "host": "localhost",
  "hotClient": Object {
    "host": "localhost",
    "logLevel": "silent",
  },
  "http2": false,
  "https": null,
  "logLevel": "silent",
  "logTime": false,
  "open": false,
  "port": 8080,
  "protocol": "http",
}
`;

exports[`app getApp + getOptions 3`] = `"Server"`;

exports[`app getApp 1`] = `
Object {
  "env": "test",
  "proxy": false,
  "subdomainOffset": 2,
}
`;

exports[`app getApp 2`] = `"Server"`;
